File: mycollab-services-pro/src/main/java/com/mycollab/pro/module/project/dao/ItemTimeLoggingMapperExt.java
package com.mycollab.pro.module.project.dao;

import com.mycollab.db.persistence.ISearchableDAO;
import com.mycollab.module.project.domain.criteria.ItemTimeLoggingSearchCriteria;
import org.apache.ibatis.annotations.Param;

/**
 * @author MyCollab Ltd
 * @since 1.0.0
 */
public interface ItemTimeLoggingMapperExt extends ISearchableDAO<ItemTimeLoggingSearchCriteria> {

    Double getTotalHoursByCriteria(@Param("searchCriteria") ItemTimeLoggingSearchCriteria criteria);
}

File: mycollab-services-pro/src/main/java/com/mycollab/pro/module/project/dao/ItemTimeLoggingMapperExt.kt
package com.mycollab.pro.module.project.dao

import com.mycollab.db.persistence.ISearchableDAO
import com.mycollab.module.project.domain.criteria.ItemTimeLoggingSearchCriteria
import org.apache.ibatis.annotations.Param

/**
 * @author MyCollab Ltd
 * @since 1.0.0
 */
interface ItemTimeLoggingMapperExt : ISearchableDAO<ItemTimeLoggingSearchCriteria> {

    fun getTotalHoursByCriteria(@Param("searchCriteria") criteria: ItemTimeLoggingSearchCriteria): Double
}
